src/test/java/seedu/address/testutil/TypicalModules.java

```diff
@@ -46,5 +55,26 @@ private TypicalModules() {
                 PROGRAMMING_METHODOLOGY_TWO,
                 DATA_STRUCTURES));
     }
+
+    /**
+     * A list of modules that affects the cap
+     * @return
+     */
+    public static List<Module> getModulesWithoutNonGradeAffectingModules() {
```

Comment:
**After some consideration, may I suggest that we stop adding to or using anything from TypicalModules.java()?**

I realize that it is very messy for each of us to keep adding our own special hardcoded test cases and expected outputs, especially when our expected outputs are highly dependent on the module parameters. 
Even if we enforce an "add only, no modification or deletion" policy for this file, this makes it impossible to refactor test cases if necessary because everyone is dependent on the same set of hard-coded modules and outputs. 

**I recommend that all test modules be built from within the component's test suite itself so all parameters are visible and 'owned' by that test suite.** Expected outputs (for example "MODULES_WITHOUT_NON_AFFECTING_MODULES_CAP) should also be created within the component's test suite. This will cause some potential duplication but that is the cost of completely isolating test suites at the file level.

Reply:
This would imply that it might need to be modified by multiple parties no? Which you didn't want initially. Like initially, typical modules was defined a certain way by Alex, then I added more modules into a typical module list.

Reply:
I think its okay to use `TypicalModules`, but at the same time we have to keep in mind that while we change something, all test would still have to pass, if changing it causes other tests to fail, then whoever changed it should make necessary changes to other test so that all test would pass. Another reason to keep this is  to think of this way: 

For example, we now want to use small case for all grades. Because we don't have a "abstract" layer, we now have to change all the grades from all the affect tests. But in the case of `TypicalModule`, I can just change it and all codes are now referencing the same object.

I think that if the same constant is referenced in multiple unit tests, we should allow it to be kept in `TypicalModule` as long as it make sense to be in it?

Reply:
I think for my case, I require a special case because I need to know before hand what is the expected CAP score of it, thus I cannot use the typical one (Typical list of modules might change). We can still maintain a typical one that should be treated by every other functions as truly typical.

Reply:
So is this resolved?

Reply:
Actually I know I merged this already but I would appreciate if you could still do another PR to factor out `MODULES_WITHOUT_NON_AFFECTING_MODULES_CAP`, `getModulesWithoutNonGradeAffectingModules` and `getModulesWithNonGradeAffectingModules` into `TranscriptTest` itself, as discussed above.